lib/esformatter.js

```diff
@@ -41,16 +41,19 @@ exports.unregister = plugins.unregister;
 
 
 function format(str, opts) {
+  _options.set(opts);
   str = plugins.stringBefore(str);
   var ast = rocambole.parse(str);
-  transform(ast, opts);
+  transform(ast, opts, true /*options already set, so skip setting there here*/);
```

Comment:
ahh, forget about the `plugins.setOptions(_curOpts.plugins)`... it works this way because it needs to **pass all the options to the plugins** - maybe we should have added a comment there explaining why.